src/lib/NotionSDK.ts

```diff
@@ -1058,49 +1126,66 @@ export class NotionSDK implements INotionSDK {
         }
     }
 
-    public async searchDatabases(
-        token: string
-    ): Promise<Array<IDatabase> | Error> {
+    public async recursiveSearchDatabases(token: string, startCursor?: string): Promise<Array<IDatabase>> {
         try {
-            const response = await this.http.post(NotionApi.SEARCH, {
-                data: {
-                    filter: {
-                        value: NotionObjectTypes.DATABASE,
-                        property: NotionObjectTypes.PROPERTY,
+            let response;
+    
+            if (startCursor) {
+                response = await this.http.post(NotionApi.SEARCH, {
+                    data: {
+                        filter: {
+                            value: NotionObjectTypes.DATABASE,
+                            property: NotionObjectTypes.PROPERTY,
+                        },
+                        start_cursor: startCursor,
                     },
-                },
-                headers: {
-                    Authorization: `Bearer ${token}`,
-                    "Content-Type": NotionApi.CONTENT_TYPE,
-                    "User-Agent": NotionApi.USER_AGENT,
-                    "Notion-Version": this.NotionVersion,
-                },
-            });
+                    headers: this.getNotionApiHeaders(token),
+                });
+            } else {
+                response = await this.http.post(NotionApi.SEARCH, {
+                    data: {
+                        filter: {
+                            value: NotionObjectTypes.DATABASE,
+                            property: NotionObjectTypes.PROPERTY,
+                        },
+                    },
+                    headers: this.getNotionApiHeaders(token),
+                });
+            }
 
             if (!response.statusCode.toString().startsWith("2")) {
-                return this.handleErrorResponse(
-                    response.statusCode,
-                    `Error While Searching Databases: `,
-                    response.content
-                );
+                throw new AppsEngineException(`Error While Searching Databases: ${response.content}`);
```

Comment:
```suggestion
                return this.handleErrorResponse(
                    response.statusCode,
                    `Error While Searching Databases: `,
                    response.content
                );
```

-  never throw an error for an usual error response, we are handling this error to later don't open a modal if we don't get the database